### ISSUE TITLE
Google speech recognition language not setup correctly for English

### ISSUE THREAD

**Issue by macavalon** [api-ai/api-ai-android-sdk#18]
The supported languages are defined in

ailib/src/main/java/ai/api/AIConfiguration.java

         /**
             * Currently supported languages
             */
            public enum SupportedLanguages {
                English("en"),

        ...
         public static SupportedLanguages fromLanguageTag(final String languageTag) {
                    switch (languageTag) {
                       case "en":
                            return English;

The language string does not correctly setup the google recognizer, taking into account country accent i.e. en-US vs en-GB

The android specification for EXTRA_LANGUAGE
http://developer.android.com/reference/android/speech/RecognizerIntent.html#EXTRA_LANGUAGE
shows that it should be Optional IETF language tag (as defined by BCP 47), for example "en-US"

I suggest adding additional enums for atleast "en-US" and "en-GB"

I've tested replacing "en" with "en_GB" in the following code
i.e. final String language = "en_GB";

ailib/src/main/java/ai/api/services/GoogleRecognitionServiceImpl.java

        private Intent createRecognitionIntent() {
                final Intent sttIntent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);
                sttIntent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL,
                        RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);
        
                final String language = config.getLanguage().replace('-', '_');
        
                sttIntent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, language);
                sttIntent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_PREFERENCE, language);

and I get much better speech recognition for my uk accent once this has been implemented.

**Comment by xVir** [comment]
Hi, @macavalon 
Yes, you are right. I didn't implement it because of there was no special need.

I will add additional options to the SupportedLanguages soon.

Thanks!

**Comment by macavalon** [comment]
Hi @xVir , it was very subtle to detect... I was seeing differences in speech accuracy of google now, vs my test app (using google speech recognition). But as soon as I used the full language code, the primary match was always good.

thanks for the quick reply...btw have you see recent news
http://techcrunch.com/2016/03/23/google-opens-access-to-its-speech-recognition-api-going-head-to-head-with-nuance/

this should mean, google speech recognition can now be introduced for the iOS libraries aswell.

best regards

**Comment by xVir** [comment]
Oh, great news! Thanks! :+1:

**Comment by xVir** [comment]
Hi, @macavalon 

Please try the library fix in version `1.8.2-SNAPHOT`.
You could get it using special Gradle config like:
```gradle
repositories {
    mavenLocal()
    mavenCentral()

    maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
}

dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    compile 'com.android.support:appcompat-v7:23.2.1'
    compile 'com.google.code.gson:gson:2.3'
    compile 'commons-io:commons-io:2.4'
    compile 'ai.api:sdk:1.8.2-SNAPSHOT@aar'
}
```

**Comment by macavalon** [comment]
Hi @xVir I've synced your commits, rebuilt my local file and tested the sample app after adding in your new language options , and all works well i.e. I get the improved speech recognition when I use en-GB
    
    public class AIServiceSampleActivity extends BaseActivity
    implements AIListener, AdapterView.OnItemSelectedListener {

    public static final String TAG = AIServiceSampleActivity.class.getName();

    final static LanguageConfig[] languages = new LanguageConfig[]{
            new LanguageConfig("en", "3485a96fb27744db83e78b8c4bc9e7b7"),
            new LanguageConfig("en-GB", "3485a96fb27744db83e78b8c4bc9e7b7"),
            new LanguageConfig("en-US", "3485a96fb27744db83e78b8c4bc9e7b7"),

thanks

**Comment by xVir** [comment]
Hi, @macavalon 
Great! Thanks for the info :+1: 
The changes will be released in nearest time.

Thanks!

**Comment by xVir** [comment]
Hi, @macavalon 
The changes released in version 1.9.0 

Thanks.
